src/ol_concourse/pipelines/open_edx/open_edx_plugins/build_publish_plugins.py

```diff
@@ -0,0 +1,139 @@
+from ol_concourse.lib.constants import REGISTRY_IMAGE
+from ol_concourse.lib.models.pipeline import (
+    AnonymousResource,
+    Command,
+    GetStep,
+    Identifier,
+    Input,
+    Job,
+    Output,
+    Pipeline,
+    Platform,
+    PutStep,
+    RegistryImage,
+    TaskConfig,
+    TaskStep,
+)
+from ol_concourse.lib.resource_types import pypi_resource
+from ol_concourse.lib.resources import git_repo, pypi
+
+plugin_dict = {
+    "src/edx_sysadmin": {
+        "target_name": "edx_sysadmin_package",
+        "pypi_package_name": "edx-sysadmin",
+    },
+    "src/ol_openedx_canvas_integration": {
+        "target_name": "canvas_integration_package",
+        "pypi_package_name": "ol-openedx-canvas-integration",
+    },
+    "src/ol_openedx_checkout_external": {
+        "target_name": "checkout_external_package",
+        "pypi_package_name": "ol-openedx-checkout-external",
+    },
+    "src/ol_openedx_course_export": {
+        "target_name": "course_export_package",
+        "pypi_package_name": "ol-openedx-course-export",
+    },
+    "src/ol_openedx_course_structure_api": {
+        "target_name": "course_structure_api_package",
+        "pypi_package_name": "ol-openedx-course-structure-api",
+    },
+    "src/ol_openedx_git_auto_export": {
+        "target_name": "edx_git_auto_export",
+        "pypi_package_name": "ol-openedx-git-auto-export",
+    },
+    "src/ol_openedx_logging": {
+        "target_name": "logging_package",
+        "pypi_package_name": "ol-openedx-logging",
+    },
+    "src/ol_openedx_otel_monitoring": {
+        "target_name": "otel_monitoring_package",
+        "pypi_package_name": "ol-openedx-otel-monitoring",
+    },
+    "src/ol_openedx_rapid_response_reports": {
+        "target_name": "rapid_response_plugin_dist",
+        "pypi_package_name": "ol-openedx-rapid-response-reports",
+    },
+    "src/ol_openedx_sentry": {
+        "target_name": "sentry_package",
+        "pypi_package_name": "ol-openedx-sentry",
+    },
+}
+
+git_resource_list = []
+pypi_resource_list = []
+job_list = []
```

Comment:
Maybe just generate a sequence of `PipelineFragment`s? Then those can be composed together, and they're easier to reference within the loop context.